sdlib/d/gc/arena.d

```diff
@@ -381,6 +420,30 @@ private:
 		registerHPD(hpd);
 	}
 
+	bool growAllocImpl(Extent* e, uint index, uint pages, uint delta) {
+		assert(mutex.isHeld(), "Mutex not held!");
+		assert(index > 0 && index <= PagesInHugePage - delta, "Invalid index!");
+		assert(pages > 0 && pages <= PagesInHugePage,
+		       "Invalid number of pages!");
+		assert(delta > 0, "Invalid delta!");
+
+		auto hpd = e.hpd;
+		unregisterHPD(hpd);
+
+		auto didGrow = hpd.set(index, delta);
+
+		if (didGrow) {
+			e.at(e.address, pages * PageSize, hpd);
+		}
+
+		if (!hpd.full) {
+			assert(!hpd.empty);
```

Comment:
Why? If this is to duplicate the in contract for registerHPD, then this is redundant. But within this function, this is a general expectation, whether the hpd is full or not, so why put it in this branch?